src/containers/daos.js

```diff
@@ -22,26 +23,41 @@ const mapDispatchToProps = (dispatch, ownProps) => {
     handleViewDaoClick: id => {
       ownProps.history.push(`/daos/${id}/proposals`)  
     },
-    joinDao: async (daoId) => {
-      // TODO: call SemadaCore.newProposal()
-      // TODO: link proposal in db to semadacore
-      // API Create Proposal
-      await dispatch(persistProposal({
-        _id: 'new',
-        daoId: daoId,
-        name: 'Join DAO',
-        evidence: '',
-        status: PROPOSAL_STATUSES.active
-      }))
+    joinDao: async (web3, semadaCore, dao) => {
+     
+      web3 = getWeb3(web3)
+      let publicAddress = await web3.eth.getCoinbase()
+      semadaCore.setProvider(web3.currentProvider)
+      let semadaCoreInstance = await semadaCore.deployed()
       
+      try {
+        let getProposalTrx = await semadaCoreInstance.getProposal(dao.proposalIndex)
```

Comment:
What are lines 34-36 intended to do? There is no proposal to retrieve before calling joinDao.